### PR TITLE
fix: don't propose on-chain signatures for existing transactions

### LINKED ISSUE #883: [Transaction Signing] Adding signatures via WalletConnect throws unexpected error
## Bug description
In a setup where a safe owns another safe.
When signing queued txs with that safe via wallet connect I always receive this error:
`Failed to send the signature. Please try again. 1337: {"nonFieldErrors":["Signature does not match sender=0x954cD69f0E902439f99156e3eeDA080752c08401. Calculated owners=['0x3819b800c67Be64029C1393c8b2e0d0d627dADE2']"]}`
In my [test safe](https://safe-web-core.staging.5afe.dev/gor:0xAecDFD3A19f777F0c03e6bf99AAfB59937d6467b/home) the sender  `0x954cD69f0E902439f99156e3eeDA080752c08401` is one of the safe’s owners.
Also interesting: Adding the signature works. So as soon as the signature is reflected by our transaction queue, the confirmation is shown in the ui and I can execute those transactions.

## Environment
 - Browser: Chrome / Firefox
 - Wallet: WalletConnect
 - Chain: Görli

## Steps to reproduce
 1. Create a transaction in a 2/n safe where one of the owners is another safe
 2. Try to sign the transaction from the other safe via wallet connect

## Expected result
The signature gets added with a success message

## Obtained result
The mentioned error.

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves #883

## How this PR fixes it

On-chain signatures may not be immediately executed, e.g. the chain is slow or it has been queued in a parent Safe. We therefore shouldn't propose them immediately and rely on our indexer to pick them up on chain instead.

## How to test it

Ensure the following interactions work: (I have confirmed they do)

#### Create/queue a transaction in a 1/n nested Safe
  - [ ] Queue approval in parent, then execute
  - [ ] Immediately approve in parent

#### Create/immediately execute a transaction of a 1/n nested Safe
  - [ ] Queue approval in parent, then execute
  - [ ] Immediately approve in parent

#### Execute a confirmed transaction of a 1/n nested Safe
  - [ ] Queue approval in parent, then execute
  - [ ] Immediately approve in parent

#### Create a transaction in a 2+/n nested Safe
  - [ ] Queue approval in parent
  - [ ] Immediately approve in parent

#### Execute a threshold - 1 confirmed transaction in a 2+/n nested Safe
  - [ ] Queue approval in parent
  - [ ] Immediately approve in parent

### CODE REVIEW

File: src/components/tx/SignOrExecuteForm/index.tsx [hunk 1]
```diff
@@ -101,17 +101,22 @@ const SignOrExecuteForm = ({
...
+    if (smartContractWallet) {
+      await dispatchOnChainSigning(createdTx, provider)
+      return
+    }
 
+    const signedTx = await dispatchTxSigning(createdTx, shouldEthSign, txId)
     const proposedTx = await dispatchTxProposal(safe.chainId, safeAddress, connectedWallet.address, signedTx, txId)
+
     return proposedTx.txId
   }
```
Comment (iamacook) [inline src/components/tx/SignOrExecuteForm/index.tsx:121]: This is the main logic. The rest is type-related.

File: src/services/tx/txSender.ts [hunk 2]
```diff
@@ -249,24 +249,23 @@ export const dispatchTxSigning = async (
 /**
  * On-Chain sign a transaction
  */
-export const dispatchOnChainSigning = async (safeTx: SafeTransaction, provider: Web3Provider) => {
+export const dispatchOnChainSigning = async (txId: string, safeTx: SafeTransaction, provider: Web3Provider) => {
   const sdkUnchecked = await getUncheckedSafeSDK(provider)
   const safeTxHash = await sdkUnchecked.getTransactionHash(safeTx)
 
-  txDispatch(TxEvent.EXECUTING, { groupKey: safeTxHash })
```
Comment (iamacook) [inline src/services/tx/txSender.ts:256]: The transaction being signed is not executing.

**Review by katspaugh — APPROVED** [review]

**Comment by francovenica** [comment]
I'm blocked in this ticket. I cannot connect with a safe as owner in WC in this pr
image

I tried in the main branch and it works fine there https://web-core.pages.dev/. I think a patch was added to main regarding WC (even tho I didn't know if it is related to the WC App)

**Comment by francovenica** [comment]
I have a tx from 2 days ago in the history tab that still shows the 2 out of 1
https://dont-propose-on-chain-signat.web-core.pages.dev/gor:0x95ddA5dc54331a1E6B65aF1Cd7167962A2f7f280/transactions/tx?id=multisig_0x95ddA5dc54331a1E6B65aF1Cd7167962A2f7f280_0x51e7aa84f6684a6616c46b301af56c089a7c1d09cc9f10c0dac5f563f2db4ff8

Also this is the latest tx, with nonce 5.
It shows that it requires 1 confirmation (makes sense, threshold was changed to 1 needed) but still has the 2 confirmations registered in the backend:
image

**Comment by francovenica** [comment]
Also I've created this ticket #930, and here, having the "Execute tx" checkbox tells me that once this modal is submitted we are sending both, the confirmation and the execution. With any other type of wallet (MM, ledger, trezor, etc) the modal doesn't have the "Execute now" because that submition only sends the "execute"

**Comment by katspaugh** [comment]
I'll merge this PR as it contains a useful refactoring that blocks other PRs.
Will extract the unindexed signature into another bug report.

**Pull request merged**